dyn_em/module_diffusion_em.F

```diff
@@ -8146,15 +8160,21 @@ SUBROUTINE vertical_diffusion_implicit(ru_tendf, rv_tendf, rw_tendf, rt_tendf,&
     DO  i = i_start, i_end
         DO  k = kts, ktf-1
            rdz_w = - g/dnw(k)/(c1h(k)*mu(i,j)+c2h(k))
-           beta  =   1.5*sqrt(tke(i,k,j))/l_diss(i,k,j)
+           IF (l_diss(i,k,j) .ne. 0) THEN
+               beta  =   1.5*sqrt(tke(i,k,j))/l_diss(i,k,j)
+           ELSE
+               beta  =   0.
+           ENDIF
            a(k)  = - 2.0*xkxavg(i,k,j)*dt*rdz_w*rdz(i,k,j)
            b(k)  =   1.0 + 2.0*dt*rdz_w*(rdz(i,k,j)*xkxavg(i,k,j)    &
                    + rdz(i,k+1,j)*xkxavg(i,k+1,j))                   &
                    + dt*beta
            c(k)  = - 2.0*xkxavg(i,k+1,j)*dt*rdz(i,k+1,j)*rdz_w
  
-           d(k)  =   tke(i,k,j)                                  &
-                   + 0.5*dt*tke(i,k,j)**1.5/l_diss(i,k,j)
+           d(k)  =   tke(i,k,j)
+           IF (l_diss(i,k,j) .ne. 0) THEN
+               d(k) = d(k) + 0.5*dt*tke(i,k,j)**1.5/l_diss(i,k,j)
+           ENDIF
```

Comment:
same story as above

Reply:
beta indicates the dissipation of TKE. When the length scale of dissipation (l_diss) is zero, the dissipation is zero. I agree with the fix.